modules/FormModel.js

```diff
@@ -113,86 +115,240 @@
+     * @method getMapping
+     * @param alias {String} the alias of the mapping - either a model mapping or a computed mapping
+     * @return the mapping config for that alias
      */
-    addModel: function(modelConfig, copy) {
-      this.__modelConfigs.push(modelConfig);
-      if (copy) {
-        this.__copyFields(modelConfig.fields, this, modelConfig.model);
-        this.__updateCache(modelConfig.model);
+    getMapping: function(alias) {
+      return this.__currentMappings[alias];
+    },
+
+    /**
+     * @method getMappings
+     * @return all the current mapping configs
+     */
+    getMappings: function() {
+      return this.__currentMappings;
+    },
+
+    /**
+     * Define or redefine how the form model pull/pushes or otherwise tracks properties between an object model(s).
+     * Examples:
+     * this.setMapping('modelAlias', true, optional model instance);
+     * this.setMapping('modelAlias, 'foo bar baz', optional model instance);
+     * this.setMapping('computedAlias', {
+     *   model1: 'foo',
+     *   model2: 'bar',
+     *   push: function(models) {
+     *     models.model1.set('foo', this.get('foobar')[0]);
+     *     models.model2.set('bar', this.get('foobar')[1]);
+     *   },
+     *   pull: function(models) {
+     *     this.set('foobar', [models.model1.foo, models.model2.bar]);
+     *   },
+     * }, optional model map)
+     * @method setMapping
+     * @param alias {String} the name for the mapping - either a model mapping or a computed mapping
+     * @param mapping {String, Boolean or Object} Provides the mapping for this alias. If trying to map to a model, then either provide
+     *  a space delimited list of fields to track as a String or the boolean true to track all the model's fields. If the mapping is for
+     *  a computed value, then provide a map from model alias to model mapping for all the fields needed for the computed and a pull method
+     *  if you want to change/combine/split object model properties before bringing them into the form model and a push method if you want to
+     *  change/combine/split form model properties before pushing them to the object models.
+     * @param [models] {Object or Backbone.Model instance} Provides instances to use for this mapping. If mapping is a computed,
+     *   provide a map from alias to model instance. If mapping is for a single model, just provide the model instance for that alias.
+     * @param [copy=false] if true, will pull values definined by this mapping after setting the mapping. Requires models to be passed in.
+     */
+    setMapping: function(alias, mapping, models, copy) {
+      var computed, fields,
+        config = {};
+      if (_.isString(mapping)) {
+        fields = mapping.split(' ');
+      } else if (mapping === true) {
+        fields = undefined;
+      } else if (_.isObject(mapping)) {
+        mapping = _.clone(mapping);
```

Comment:
it's a shallow copy and i don't believe it matters. deep cloning things with functions is more trouble then i expect it to be for others.